Ds3/Calls/DeleteFolderRequest.cs

```diff
@@ -0,0 +1,70 @@
+﻿/*
+ * ******************************************************************************
+ *   Copyright 2014 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+using System.Net;
+
+using Ds3.Models;
+
+namespace Ds3.Calls
+{
+    public class DeleteFolderRequest : Ds3Request
+    {
+        private string _bucketId;
+        public string BucketId
+        {
+            get { return _bucketId; }
+            set { WithBucketId(value); }
+        }
+        protected DeleteFolderRequest WithBucketId(string bucket)

```

Comment:
Gotcha – do I do another pull request?

jk

From: Ryan Moore [mailto:notifications@github.com]
Sent: Tuesday, September 08, 2015 11:27 AM
To: SpectraLogic/ds3_net_sdk <ds3_net_sdk@noreply.github.com>
Cc: John Kranz <johnk@spectralogic.com>
Subject: Re: [ds3_net_sdk] jk -- Add DeleteFolder Call and unit Tests (#45)


In Ds3/Calls/DeleteFolderRequest.cs<https://github.com/SpectraLogic/ds3_net_sdk/pull/45#discussion_r38954161>:

> +

> +using System.Net;

> +

> +using Ds3.Models;

> +

> +namespace Ds3.Calls

> +{

> +    public class DeleteFolderRequest : Ds3Request

> +    {

> +        private string _bucketId;

> +        public string BucketId

> +        {

> +            get { return _bucketId; }

> +            set { WithBucketId(value); }

> +        }

> +        protected DeleteFolderRequest WithBucketId(string bucket)

In other sdks, when we have a required arg, we don't provide a mutator for it. See https://github.com/SpectraLogic/ds3_net_sdk/blob/master/Ds3/Calls/GetJobRequest.cs for an example of what I mean

—
Reply to this email directly or view it on GitHub<https://github.com/SpectraLogic/ds3_net_sdk/pull/45/files#r38954161>.